src/Amalgam/evaluablenode/EvaluableNodeTreeFunctions.h

```diff
@@ -58,35 +58,54 @@ EntityReferenceType TraverseToExistingEntityReferenceViaEvaluableNodeIDPath(Enti
 
 	auto &ocn = id_path->GetOrderedChildNodes();
 
-	//always keep one to two locks active at once to walk down the entity containers
-	EntityReadReference relative_entity_read(container);
+	//size of the entity list excluding trailing nulls
+	size_t non_null_size = ocn.size();
+	for(; non_null_size > 0; non_null_size--)
+	{
+		if(!EvaluableNode::IsNull(ocn[non_null_size - 1]))
+			break;
+	}
 
-	if(ocn.size() == 0)
+	if(non_null_size == 0)
 	{
+		//if empty list, return the container itself
+		if(id_path->GetType() == ENT_LIST)
+			return EntityReferenceType(container);
+
 		//if the string doesn't exist, then there can't be an entity with that name
 		StringInternPool::StringID sid = EvaluableNode::ToStringIDIfExists(id_path);
 		return EntityReferenceType(container->GetContainedEntity(sid));
 	}
 
-	size_t last_ce = ocn.size();
-	for(size_t i = 0; i < last_ce; i++)
+	//if traversing, then it needs to be a list, otherwise not a valid entity
+	if(id_path->GetType() != ENT_LIST)
+		return EntityReferenceType(nullptr);
+
+	//always keep one to two locks active at once to walk down the entity containers
```

Comment:
Not sure I understand this comment. What does "keep one to two locks active" mean in this context?